### PR TITLE
chore(deploy): copilot renaming for deploy pkg

### PR DESCRIPTION
**Pull request by iamhopaul123**
Copilot renaming for `deploy` package.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/app.go [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
-// DeleteAppInput holds the fields required to delete an application.
-type DeleteAppInput struct {
-	AppName     string
-	EnvName     string
-	ProjectName string
+// DeleteServiceInput holds the fields required to delete a service.
+type DeleteServiceInput struct {
+	Name    string // Name of the service that needs to be deleted.
+	EnvName string // Name of the environment that service deployed in.
+	AppName string // Name of the application that service belongs to.
```
Comment (efekarakus) [inline internal/pkg/deploy/app.go:12]: nit:
```suggestion
	EnvName string // Name of the environment the service is deployed in.
	AppName string // Name of the application the service belongs to.
```

File: internal/pkg/deploy/env.go [hunk 2]
```diff
@@ -1,21 +1,21 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// Package deploy holds the structures to deploy applications and environments.
+// Package deploy holds the structures to deploy infrastructure resources.
 // This file defines environment deployment resources.
 package deploy
```
Comment (efekarakus) [inline internal/pkg/deploy/env.go:6]: unrelated: just fyi, my dream state is to completely get rid of all these structs in deploy and have something like this

**hierarchy**
```
deploy/
   cloudformation/
   cdk/
   deploy.go
```

and all these subpackages implement the interface described in deploy.go:
```go
// in deploy.go
type Deployer interface {
    Deploy() error
}
```
for example, in cloudformation/svc.go:
```go
type ServiceDeployer struct {
// service data
}
func (d *ServiceDeployer) Deploy() error { }
```

File: internal/pkg/deploy/app.go [hunk 3]
```diff
@@ -2,12 +2,12 @@
 // SPDX-License-Identifier: Apache-2.0
```
Comment (efekarakus) [inline internal/pkg/deploy/app.go:2]: Do we need to rename these files?

File: internal/pkg/deploy/pipeline_test.go [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// Package deploy holds the structures to deploy applications and environments.
+// Package deploy holds the structures to deploy infrastructure resources.
```
Comment (efekarakus) [inline internal/pkg/deploy/pipeline_test.go:4]: we can just remove this comment

File: internal/pkg/deploy/app.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/deploy/app.go:2]: yes I forgot

File: internal/pkg/cli/project_init_test.go [hunk 5]
```diff
@@ -299,8 +299,8 @@ func TestInitProjectOpts_Execute(t *testing.T) {
 					Create(gomock.Eq("project")).Return(nil)
 				mockProgress.EXPECT().Start(fmt.Sprintf(fmtDeployProjectStart, "project"))
 				mockDeployer.EXPECT().
-					DeployProject(&deploy.CreateProjectInput{
-						Project:    "project",
+					DeployProject(&deploy.CreateAppInput{
+						Name:       "project",
```
Comment (kohidave) [inline internal/pkg/cli/project_init_test.go:303]:
```suggestion
						Name:       "app",
```

File: internal/pkg/deploy/cloudformation/env.go [hunk 6]
```diff
@@ -41,7 +41,7 @@ func (cf CloudFormation) StreamEnvironmentCreation(env *deploy.CreateEnvironment
 // DeleteEnvironment deletes the CloudFormation stack of an environment.
 func (cf CloudFormation) DeleteEnvironment(projectName, envName string) error {
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/env.go:42]:
```suggestion
func (cf CloudFormation) DeleteEnvironment(appName, envName string) error {
```

File: internal/pkg/deploy/cloudformation/env.go [hunk 7]
```diff
@@ -41,7 +41,7 @@ func (cf CloudFormation) StreamEnvironmentCreation(env *deploy.CreateEnvironment
...
 	conf := stack.NewEnvStackConfig(&deploy.CreateEnvironmentInput{
-		Project: projectName,
+		AppName: projectName,
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/env.go:44]:
```suggestion
		AppName: appName,
```

File: internal/pkg/deploy/cloudformation/env.go [hunk 8]
```diff
@@ -70,7 +70,7 @@ func (cf CloudFormation) streamEnvironmentResponse(done chan struct{}, resp chan
 // GetEnvironment returns the Environment metadata from the CloudFormation stack.
 func (cf CloudFormation) GetEnvironment(projectName, envName string) (*archer.Environment, error) {
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/env.go:71]:
```suggestion
func (cf CloudFormation) GetEnvironment(appName, envName string) (*archer.Environment, error) {
```

File: internal/pkg/deploy/cloudformation/env.go [hunk 9]
```diff
@@ -70,7 +70,7 @@ func (cf CloudFormation) streamEnvironmentResponse(done chan struct{}, resp chan
...
 	conf := stack.NewEnvStackConfig(&deploy.CreateEnvironmentInput{
-		Project: projectName,
+		AppName: projectName,
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/env.go:73]:
```suggestion
		AppName: appName,
```

File: internal/pkg/deploy/cloudformation/project.go [hunk 10]
```diff
@@ -21,7 +21,7 @@ import (
 // environment, such as ECR Repos, CodePipeline KMS keys & S3 buckets.
 // We deploy project resources through StackSets - that way we can have one
 // template that we update and all regional stacks are updated.
-func (cf CloudFormation) DeployProject(in *deploy.CreateProjectInput) error {
+func (cf CloudFormation) DeployProject(in *deploy.CreateAppInput) error {
 	projectConfig := stack.NewAppStackConfig(in)
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/project.go:25]: Can you rename all the project variables into appConfig, too?

File: internal/pkg/deploy/cloudformation/stack/env.go [hunk 11]
```diff
@@ -155,7 +155,7 @@ func (e *EnvStackConfig) ToEnv(stack *cloudformation.Stack) (*archer.Environment
 
 	return &archer.Environment{
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env.go:156]: Are we going to tackle this later, moving to `config.Environment`? I think that's fine, just wanted to make sure.

File: internal/pkg/deploy/cloudformation/stack/env_test.go [hunk 12]
```diff
@@ -211,7 +211,7 @@ func TestEnvTags(t *testing.T) {
 	env := &EnvStackConfig{
 		CreateEnvironmentInput: &deploy.CreateEnvironmentInput{
 			Name:    "env",
-			Project: "project",
+			AppName: "project",
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env_test.go:214]:
```suggestion
			AppName: "app",
```

File: internal/pkg/deploy/cloudformation/stack/env_test.go [hunk 13]
```diff
@@ -211,7 +211,7 @@ func TestEnvTags(t *testing.T) {
...
 			AdditionalTags: map[string]string{
 				"owner":   "boss",
 				AppTagKey: "overrideproject",
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env_test.go:217]:
```suggestion
				AppTagKey: "overrideapp",
```

File: internal/pkg/deploy/cloudformation/stack/env_test.go [hunk 14]
```diff
@@ -307,7 +307,7 @@ func mockEnvironmentStack(stackArn, managerRoleARN, executionRoleARN string) *cl
 func mockDeployEnvironmentInput() *deploy.CreateEnvironmentInput {
 	return &deploy.CreateEnvironmentInput{
 		Name:                     "env",
-		Project:                  "project",
+		AppName:                  "project",
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env_test.go:310]:
```suggestion
		AppName:                  "app",
```

File: templates/cicd/pipeline_cfn.yml [hunk 15]
```diff
@@ -6,7 +6,7 @@
 # CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 AWSTemplateFormatVersion: '2010-09-09'
-Description: CodePipeline for the {{$.ProjectName}}
+Description: CodePipeline for the {{$.AppName}}
```
Comment (kohidave) [inline templates/cicd/pipeline_cfn.yml:9]:
```suggestion
Description: CodePipeline for {{$.AppName}}
```

File: internal/pkg/cli/project_init_test.go (hunk 5 above)
Comment (iamhopaul123) [reply internal/pkg/cli/project_init_test.go:303]: I prefer to do this kind of change when we are renaming the `cli` package. The good thing about this approach is that we won't get confused when we refactoring the `cli` package (we'll know "app" means the previous app but not "project").

**Pull request merged**